patches/nginx-1.17.8-set_uri_security_issue.patch

```diff
@@ -0,0 +1,39 @@
+diff --git a/src/http/modules/ngx_http_static_module.c b/src/http/modules/ngx_http_static_module.c
+index 282d6ee..cb5b6c1 100644
+--- a/src/http/modules/ngx_http_static_module.c
++++ b/src/http/modules/ngx_http_static_module.c
+@@ -44,6 +44,20 @@ ngx_module_t  ngx_http_static_module = {
+     NGX_MODULE_V1_PADDING
+ };
+ 
++static ngx_int_t
++ngx_http_check_safe_location(u_char *str, size_t len)
++{
++    u_char ch;
++    size_t i;
++
++    for (i = 0; i < len; i++) {
++        ch = str[i];
++        if (ch <= 31 || ch == 127) {
++            return NGX_ERROR;
```

Comment:
Maybe we should log the bad uri here instead so that we know exactly what bad character it is.